src/libraries/Microsoft.Extensions.Hosting/src/Internal/HostingLoggerExtensions.cs

```diff
@@ -2,19 +2,21 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
+using System.Diagnostics;
 using System.Reflection;
 using Microsoft.Extensions.Logging;
 
 namespace Microsoft.Extensions.Hosting.Internal
 {
     internal static class HostingLoggerExtensions
     {
-        public static void ApplicationError(this ILogger logger, EventId eventId, string message, Exception exception)
+        public static void ApplicationError(this ILogger logger, EventId eventId, string? message, Exception? exception)
         {
             if (exception is ReflectionTypeLoadException reflectionTypeLoadException)
             {
-                foreach (Exception ex in reflectionTypeLoadException.LoaderExceptions)
+                foreach (Exception? ex in reflectionTypeLoadException.LoaderExceptions)
                 {
+                    Debug.Assert(ex != null);
```

Comment:
Since `ReflectionTypeLoadException.LoaderExceptions` can contain nulls, (see [https://docs.microsoft.com/dotnet/api/system.reflection.reflectiontypeloadexception.loaderexceptions](https://docs.microsoft.com/dotnet/api/system.reflection.reflectiontypeloadexception.loaderexceptions)), I think it would be better to skip the nulls and not append a message if `ex is null`. That way you don't get a NRE here.